File: src/main/java/infinitespire/relics/BurningSword.java
package infinitespire.relics;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.status.Burn;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import com.megacrit.cardcrawl.relics.AbstractRelic;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.Relic;
import infinitespire.powers.CriticalPower;

public class BurningSword extends Relic {

	public static final String ID = InfiniteSpire.createID("Burning Sword");
	public static int attacksPlayed = 0;
	
	public BurningSword() {
		super(ID, "burningsword", RelicTier.BOSS, LandingSound.MAGICAL);
		attacksPlayed = 0;
	}

	@Override
	public AbstractRelic makeCopy() {
		return new BurningSword();
	}

	@Override
	public void atTurnStart() {
		attacksPlayed = 0;
		AbstractDungeon.actionManager.addToBottom(
			new ApplyPowerAction(
				AbstractDungeon.player,
				AbstractDungeon.player,
				new CriticalPower(AbstractDungeon.player), 1));
	}

	@Override
	public void onPlayCard(AbstractCard c, AbstractMonster m) {
		if(c.type == AbstractCard.CardType.ATTACK && attacksPlayed == 0){
			AbstractCard burn = new Burn();
			burn.upgrade();
			AbstractDungeon.actionManager.addToBottom(new MakeTempCardInHandAction(burn));
			attacksPlayed++;
		}
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/BurningSword.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.powers.CriticalPower
import com.blanktheevil.infinitespire.relics.abstracts.Relic
import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.status.Burn
import com.megacrit.cardcrawl.monsters.AbstractMonster

class BurningSword : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "Burning Sword".makeID()
    private val IMG = "burningsword"
    private val TIER = RelicTier.BOSS
    private val SOUND = LandingSound.MAGICAL
  }

  var attacksPlayed = 0

  override fun atTurnStart() {
    attacksPlayed = 0
    player.applyPower(CriticalPower(player))
  }

  override fun onPlayCard(card: AbstractCard?, monster: AbstractMonster?) {
    if(card != null && card.type == AbstractCard.CardType.ATTACK && attacksPlayed == 0) {
      addToBot(
        MakeTempCardInHandAction(Burn())
      )
      attacksPlayed++
    }
  }
}